src/views/CreateType.vue

```diff
@@ -0,0 +1,222 @@
+    <ViewHeader
+      :title="`Choose a parent type for ${newResourceType.name}`"
+      :backLink="{ onClick: previousStep }"
+    />
+    <ListSection :list="parentTypeList"/>
+  </main>
+
+  <main v-else-if="formState === 3">
+    <ViewHeader
+      :title="`Define attributes for ${newResourceType.name}`"
+      :backLink="{ onClick: previousStep }"
+    />
+    <ListSection :title="`Attributes of ${parentType.name}`" :list="parentTypeAttributeList"/>
+
+    <ListSection :title="`Attributes of ${newResourceType.name}`">
+      <Li v-for="attribute in attributeListLeft" :key="attribute.id" :options="attribute"/>
+
+      <Li v-if="currentlyEditingAttribute >= 0" class="attribute-form">
+        <h2>New Attribute</h2>
+        <Input
+          :value.sync="editingAttribute.name"
+          name="Name"
+          placeholder="Location"
+          :autofocus="true"
+          :required="true"
+        />
+        <Select
+          :value.sync="editingAttribute.dataType"
+          name="Data Type"
+          :required="true">
+          <option>string</option>
+          <option>number</option>
+          <option>boolean</option>
+        </Select>
+        <Toggle :value.sync="editingAttribute.required" name="Required"/>
+        <Button text="Save Attribute" :onClick="saveAttribute"/>
+      </Li>
+
+      <Li v-for="attribute in attributeListRight" :key="attribute.id" :options="attribute"/>
+      <Li :options="addButtonOptions"/>
+    </ListSection>
+
+    <Button text="Create Resource Type" :onClick="createResourceType"/>
+  </main>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'vue-property-decorator';
+import {
+  ResourceTypes,
+  ResourceType,
+  ResourceTypeAttribute,
+} from '@/apis/rembrandt/rembrandt';
+import Li, { LiOptions } from '@/components/Li.vue';
+import ViewHeader from '@/components/ViewHeader.vue';
+import ListSection from '@/components/ListSection.vue';
+import Input from '@/components/Input.vue';
+import Toggle from '@/components/Toggle.vue';
+import Button from '@/components/Button.vue';
+import Select from '@/components/Select.vue';
+import Utils from '@/utils/Utils';
+
+@Component({
+  components: {
+    ListSection,
+    Li,
+    ViewHeader,
+    Input,
+    Toggle,
+    Button,
+    Select,
+  },
+})
+export default class Types extends Vue {
+  // region public static methods
+  public static emptyResourceType(): ResourceType {
+    return {
+      name: '',
+      parentType: '',
+      abstract: false,
+      attributes: [],
+    };
+  }
+
+  public static emptyResourceTypeAttribute(): ResourceTypeAttribute {
+    return {
+      name: '',
+      dataType: 'string',
+      required: true,
+    };
+  }
+  // endregion
+
+  // region private static methods
+  // endregion
+
+  // region public members
+  public resourceTypes: ResourceType[] = [];
+  public newResourceType: ResourceType;
+  public formState: number = 0;
+  public currentlyEditingAttribute: number = -1;
```

Comment:
initialized at -1?

Reply:
This is default js behavior of Array.findIndex. 
https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex